### PR TITLE
ZTS: Update deprecated Github Action version numbers

### PR DESCRIPTION
**Pull request by mcmilk**
### Motivation and Context
GitHub Actions is transitioning from Node 16 to Node 20.

So we need to update these:
actions/checkout@v3 -> v4
actions/download-artifact@v3 -> v4
actions/upload-artifact@v3 -> v4

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by andrewc12 — APPROVED** [review]

**Review by gmelikov — APPROVED** [review]

**Comment by mcmilk** [comment]
A fix for the upload is needed, I will force push some changes later.

File: .github/workflows/README.md [hunk 1]
```diff
@@ -4,44 +4,49 @@
...
+  Part4-20.04-->CleanUp+Summary
+  Part1-22.04-->CleanUp+Summary
+  Part2-22.04-->CleanUp+Summary
+  Part3-22.04-->CleanUp+Summary
+  Part4-22.04-->CleanUp+Summary
+  sanity-checks-22.04-->CleanUp+Summary
 end
 
 subgraph Functional Testings
+  sanity-checks-20.04
```
Comment (andrewc12) [inline .github/workflows/README.md:20]: Should these be in this subgraph?

Comment (mcmilk) [reply .github/workflows/README.md:20]: Yes, the results of them go into the `CleanUp+Summary` thing in the end.

Should I put it into the `Functional Testing` area and create sth. like a `Sanity-Result-20.04` which then `CleanUp+Summary`?

**Review by behlendorf — APPROVED** [review]

**Comment by andrewc12** [comment]
The generated graph seems broken.
I'll do a pr.

**Comment by andrewc12** [comment]
Current graph
github com_andrewc121

My proposal
github com_andrewc122

Is this better?

**Pull request merged**